geode-core/src/test/java/org/apache/geode/internal/cache/tier/sockets/ClientUpdateMessageImplTest.java

```diff
@@ -109,4 +135,96 @@ private ClientUpdateMessageImpl getTestClientUpdateMessage() {
+    ClientProxyMembershipID[] clients = new ClientProxyMembershipID[numOfEvents];
+    prepareCqInfo(numOfEvents, cqEvents, cqNames, clients);
+
+    addClientCqConcurrently(clientUpdateMessageImpl, numOfEvents, cqEvents, cqNames, clients);
+
+    assertThat(clientUpdateMessageImpl.getClientCqs().size()).isEqualTo(2);
+    assertThat(clientUpdateMessageImpl.getClientCqs().get(client1)).isInstanceOf(
+        ClientUpdateMessageImpl.CqNameToOpHashMap.class);
+    ClientUpdateMessageImpl.CqNameToOpHashMap client1Cqs =
+        (ClientUpdateMessageImpl.CqNameToOpHashMap) clientUpdateMessageImpl.getClientCqs()
+            .get(client1);
+    for (int i = 0; i < 3; i++) {
+      assertThat(client1Cqs.get("cqName" + i)).isEqualTo(i);
+    }
+
+    assertThat(clientUpdateMessageImpl.getClientCqs().get(client2)).isInstanceOf(
+        ClientUpdateMessageImpl.CqNameToOpSingleEntry.class);
+    ClientUpdateMessageImpl.CqNameToOpSingleEntry client2Cqs =
+        (ClientUpdateMessageImpl.CqNameToOpSingleEntry) clientUpdateMessageImpl.getClientCqs()
+            .get(client2);
+    assertThat(client2Cqs.isEmpty()).isFalse();
+  }
+
+  private void prepareCqInfo(int numOfEvents, int[] cqEvents, String[] cqNames,
+      ClientProxyMembershipID[] clients) {
+    for (int i = 0; i < numOfEvents; i++) {
+      cqEvents[i] = i;
+      cqNames[i] = "cqName" + i;
+      if (i < 3) {
+        clients[i] = client1;
+      } else {
+        clients[i] = client2;
+      }
+    }
+  }
+
+  private void addClientCqConcurrently(ClientUpdateMessageImpl clientUpdateMessageImpl,
+      int numOfEvents, int[] cqEvents, String[] cqNames, ClientProxyMembershipID[] clients)
+      throws InterruptedException, java.util.concurrent.ExecutionException {
+    List<Future<Void>> futures = new ArrayList<>();
+    for (int i = 0; i < numOfEvents; i++) {
+      ClientProxyMembershipID client = clients[i];
+      String cqName = cqNames[i];
+      int cqEvent = cqEvents[i];
+      futures.add(executorService
+          .submit(() -> clientUpdateMessageImpl.addClientCq(client, cqName, cqEvent)));
+    }
+    for (Future<Void> future : futures) {
+      future.get();
+    }
+  }
+
+  @Test
+  public void addOrSetClientCqsCanSetIfCqsMapIsNull() {
+    ClientUpdateMessageImpl clientUpdateMessageImpl = new ClientUpdateMessageImpl();
+
+    clientUpdateMessageImpl.addOrSetClientCqs(client1, clientCqs);
+
+    assertThat(clientUpdateMessageImpl.getClientCqs()).isEqualTo(clientCqs);
+  }
+
+  @Test
+  public void addOrSetClientCqsCanAddCqsIfCqsMapNotNull() {
+    ClientUpdateMessageImpl clientUpdateMessageImpl = new ClientUpdateMessageImpl();
+    ClientProxyMembershipID clientProxyMembershipID = mock(ClientProxyMembershipID.class);
+    clientUpdateMessageImpl.addClientCq(clientProxyMembershipID, "cqName", 10);
+
+    clientUpdateMessageImpl.addOrSetClientCqs(client1, clientCqs);
+
+    assertThat(clientUpdateMessageImpl.getClientCqs().size()).isEqualTo(2);
```

Comment:
`ClientUpdateMessageImpl.getClientCqs()` returns a map, so it would be better to use the `Map` assertions everywhere instead of `int` or `boolean` assertions:
```
assertThat(clientUpdateMessageImpl.getClientCqs()).hasSize(2);
```